Support i8 type

nandorKollar opened the pull request:
Problem

Compilation failure with i8 type

Solution

Introduce i8 type identifier as an alias for byte type.

Result

mosesn commented:
this seems reasonable to me, thanks for the patch!!  how would you feel about adding something to one of the goldfile tests that validates that this works?

nandorKollar commented:
@mosesn sure I can! Just committed a change related to gold files, did you think of these tests?

mosesn commented:
LGTM!  @yufangong do you want to take a look here before we pull this in?

yufangong commented:
this will fix https://github.com/twitter/scrooge/issues/309

yufangong commented on scrooge-generator-tests/src/test/resources/gold_file_output_cocoa/TFNTwitterThriftGoldRequest.h:
```diff
@@ -1,8 +1,8 @@
- *   version: ?
- *   rev: ?
- *   built at: ?
+ *   version: 19.4.0-SNAPSHOT
```
Can we revert these?

yufangong commented on scrooge-generator-tests/src/test/scala/com/twitter/scrooge/goldfile/GoldFileTest.scala:
```diff
@@ -62,7 +62,7 @@ abstract class GoldFileTest extends FunSuite with BeforeAndAfterAll {
   protected def experimentFlags: Seq[String] = Seq.empty
   protected def language: String
-  protected def deleteTempFiles: Boolean = true
+  protected def deleteTempFiles: Boolean = false
```
revert?

nandorKollar commented:
Ouch, I thought I reverted these intermediate changes before submitting the PR, but looks like I didn't. @yufangong thanks for the review, addressed the comments.

yufangong commented:
Hi @nandorKollar, your PR has been merged here https://github.com/twitter/scrooge/commit/03d84aea40c9edf87e8c69090c5077b8779fa655. Thanks so much for your contribution.
